src/function_data.jl

```diff
@@ -399,8 +399,264 @@ QuadraticFunctionData(data::LinearFunctionData) =
 Base.convert(::Type{QuadraticFunctionData}, data::LinearFunctionData) =
     QuadraticFunctionData(data)
 
+# GET_DOMAIN
+"Get the domain of the function represented by the `LinearFunctionData` or `QuadraticFunctionData` (always `(-Inf, Inf)` for these types)."
+get_domain(::Union{LinearFunctionData, QuadraticFunctionData}) = (-Inf, Inf)
+
+"Get the domain of the function represented by the `PiecewiseLinearData`."
+get_domain(fd::PiecewiseLinearData) =
+    (first(get_points(fd)).x, last(get_points(fd)).x)  # avoiding get_x_coords to avoid extra allocation
+
+"Get the domain of the function represented by the `PiecewiseStepData`."
+get_domain(fd::PiecewiseStepData) =
+    (first(get_x_coords(fd)), last(get_x_coords(fd)))
+
+# ZERO
 "Get a `LinearFunctionData` representing the function `f(x) = 0`"
 Base.zero(::Union{LinearFunctionData, Type{LinearFunctionData}}) = LinearFunctionData(0, 0)
 
+"Get a `QuadraticFunctionData` representing the function `f(x) = 0`"
+Base.zero(::Union{QuadraticFunctionData, Type{QuadraticFunctionData}}) =
```

Comment:
why would these functions take an instance and not just a type?, I see that Julia takes `zero(1.0)`